lib/productions/helpers.js

```diff
@@ -242,3 +242,43 @@ export function findLastIndex(array, predicate) {
   }
   return array.length - index - 1;
 }
+
+/**
+ * Returns a proxy that auto-assign `parent` field.
+ * @template T
+ * @param {T} data
+ * @param {*} parent this is `data` by default
+ * @return {T}
+ */
+export function autoParenter(data, parent) {
+  if (!parent) {
```

Comment:
Might be nice to document what's happening here (and below).... this is really nice,  but you are right that it's a bit magical. I'd be a bit scared to come in here in the future if anything goes wrong.